### ISSUE TITLE
Listening Mode: Prevent "Grant Access" Flow for Non-English Languages

### ISSUE THREAD

**Issue by rhsalas3** [willowtreeapps/eyespeak-ios#672]
### Description
We have implemented a message to the user that their device does not support listening mode, when that device is set to a non-English language (issue #645 -> PR #663 ). On a fresh install of the app the user must complete the flow to grant access to Speech Recognition and the Microphone before Listening Mode is enabled. In this scenario, of a device that is set to a non-English language, they are presented with an unresponsive and unnecessary grant-access-to-Speech-Recognition screen. We should consider presenting our "device is not supported" message here as well, instead of the grant access flow. 

Furthermore, it may be a good idea to include our "when listening mode is enabled" description along with the message. This way the user can still be informed about what the feature is.

Update: Question for the team...what is the expected behavior when your device is set to an unsupported language, which defaults to English?

**Acceptance Criteria**:
- [x] While the device is set to a non-English language, the user is presented with the "device is not supported" message instead of the grant access flow on the Settings > Listening Mode screen, and Settings > Categories and Phrases > 'Listen' category details screen.
    - [x] on a fresh install of Vocable
    - [x] when upgrading from a version that does not have listening mode to one that does

- [x] ~~The "when listening mode is enabled" description text accompanies the "devices is not supported" message where the latter is displayed instead of the "grant access" flow~~. We decided not to include the description, opting for MVP whereby we only display the "device is not supported" text.

- [x] The "device is not supported" message continues to display as expected, when the user has already granted access to listening mode but then toggles their device to a non-English language; per our implementation of #645 

**Design**:

We will probably need designs or at least approved copy.

**Comment by rhsalas3** [comment]
@SashankPatel 

Here is the ticket I referenced in the comment I left on #663

**Comment by rhsalas3** [comment]
Tested and verified the fix on the following devices:
- iPhone 8 Plus, iOS 14.8.1
- iPhone X, iOS 14.2
- iPad Pro, iOS 15.3.1

There is one outstanding question I asked @jsmorgan42 in Slack; whether or not we wanted to include descriptive text of what Listening Mode while in this state. It was mentioned in the AC and we did talk about wanting it there but may need to address it later (possibly with some formal copy defined). 

I will update this ticket with any comments related to the question.
